pkg/bucketeer/api/rest.go

```diff
@@ -0,0 +1,254 @@
+	ID                   string          `json:"id,omitempty"`
+	Event                json.RawMessage `json:"event,omitempty"`
+	EnvironmentNamespace string          `json:"environment_namespace,omitempty"`
+	Type                 eventType       `json:"type,omitempty"`
+}
+
+type MetricsEvent struct {
+	Timestamp int64                  `json:"timestamp,omitempty"`
+	Event     json.RawMessage        `json:"event,omitempty"`
+	Type      metricsDetailEventType `json:"type,omitempty"`
+}
+
+type GoalEvent struct {
+	Timestamp int64        `json:"timestamp,omitempty"`
+	GoalID    string       `json:"goal_id,omitempty"`
+	UserID    string       `json:"user_id,omitempty"`
+	Value     float64      `json:"value,omitempty"`
+	User      *user.User   `json:"user,omitempty"`
+	Tag       string       `json:"tag,omitempty"`
+	SourceID  SourceIDType `json:"source_id,omitempty"`
+}
+
+type InternalErrorCountMetricsEvent struct {
+	Tag string `json:"tag,omitempty"`
+}
+
+type GetEvaluationSizeMetricsEvent struct {
+	Labels   map[string]string `json:"labels,omitempty"`
+	SizeByte int32             `json:"size_byte,omitempty"`
+}
+
+type GetEvaluationLatencyMetricsEvent struct {
+	Labels   map[string]string `json:"labels,omitempty"`
+	Duration time.Duration     `json:"duration,omitempty"`
+}
+
+type TimeoutErrorCountMetricsEvent struct {
+	Tag string `json:"tag,omitempty"`
+}
+
+type Variation struct {
+	ID          string `json:"id,omitempty"`
+	Value       string `json:"value,omitempty"` // number or even json object
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+
+type Evaluation struct {
+	ID             string  `json:"id,omitempty"`
+	FeatureID      string  `json:"feature_id,omitempty"`
+	FeatureVersion int32   `json:"feature_version,omitempty"`
+	UserID         string  `json:"user_id,omitempty"`
+	VariationID    string  `json:"variation_id,omitempty"`
+	Reason         *Reason `json:"reason,omitempty"`
+	VariationValue string  `json:"variation_value,omitempty"`
+}
+
+type EvaluationEvent struct {
+	Timestamp      int64      `json:"timestamp,omitempty"`
+	FeatureID      string     `json:"feature_id,omitempty"`
+	FeatureVersion int32      `json:"feature_version,omitempty"`
+	VariationID    string     `json:"variation_id,omitempty"`
+	User           *user.User `json:"user,omitempty"`
+	Reason         *Reason    `json:"reason,omitempty"`
+	Tag            string     `json:"tag,omitempty"`
+	SourceID       int32      `json:"source_id,omitempty"`
+}
+
+type RegisterEventsResponseError struct {
+	Retriable bool   `json:"retriable,omitempty"`
+	Message   string `json:"message,omitempty"`
+}
+
+type ReasonType int32
+
+const (
+	ReasonClient ReasonType = 4
+)
+
+type Reason struct {
+	Type   ReasonType `json:"type,omitempty"`
+	RuleID string     `json:"rule_id,omitempty"`
+}
+
+type UserEvaluationsState int32
+
+const (
+	UserEvaluationsFULL UserEvaluationsState = 2
+)
+
+func (c *client) ping() (*pingResponse, error) {
```

Comment:
Do we need this?

Reply:
I think, yes. In `NewUser`, we can check if the configuration is correct through the ping api.
https://github.com/ca-dp/bucketeer-go-server-sdk/pull/86/files#diff-093fd1368a24e3737f9fe7209600af91df5f30bd899e921ab00c3e77e6b6056dR31